### PR TITLE
allow objects returned to pool after pool_destroy()

### PR DESCRIPTION
**Pull request by ekoby**

### CODE REVIEW

File: library/pool.c [hunk 1]
```diff
@@ -31,41 +31,46 @@ struct pool_obj_s {
...
+
+    while (!LIST_EMPTY(&pool->pool)) {
+        struct pool_obj_s *m = LIST_FIRST(&pool->pool);
         LIST_REMOVE(m, _next);
         free(m);
     }
 
-    free(p);
+    if (pool->out == 0)
+        free(pool);
```
Comment (dovholuknf) [inline library/pool.c:59]: why the 'if' here? something else must eventually end up free'ing the pool?

File: library/pool.c [hunk 2]
```diff
@@ -31,41 +32,47 @@ struct pool_obj_s {
...
 }
 
 bool pool_has_available(pool_t *pool) {
-    return !LIST_EMPTY(&pool->pool) || pool->count > 0;
+    assert(!pool->is_closed);
```
Comment (smilindave26) [inline library/pool.c:64]: that's pretty violent.  better to do WTF log?

File: library/pool.c [hunk 3]
```diff
@@ -31,41 +32,47 @@ struct pool_obj_s {
...
+    return !LIST_EMPTY(&pool->pool) || pool->capacity > pool->out;
 }
 
 void *pool_alloc_obj(pool_t *pool) {
+    assert(!pool->is_closed);
```
Comment (smilindave26) [inline library/pool.c:69]: more violence!

File: library/pool.c (hunk 2 above)
Comment (ekoby) [reply library/pool.c:64]: this flags a programming error, i.e. should never happen

**Pull request merged**